### PR TITLE
Allow <clinit> special java method too

### PR DESCRIPTION
**Pull request by Hubbitus**
It supported many time: http://docs.oracle.com/javase/specs/jvms/se7/html/jvms-2.html#jvms-2.9

### CODE REVIEW

**Comment by purbon** [comment]
Good morning @Hubbitus, thanks a lot for your contribution. In order to move forward with your contribution you will need to sign the CLA, more info at -> https://www.elastic.co/contributor-agreement

On the other side, we just added the option to test the grok patterns we've in here, adding test to validate your changes it will be cool to merge this in too. You can take inspiration to write the test from (https://github.com/logstash-plugins/logstash-patterns-core/blob/master/spec/patterns/syslog_spec.rb), don't heasitate to ask any question you might have, we'll be happy to provide help.

**Comment by Hubbitus** [comment]
I have signed CLA now. Nut details link lead to 404 error (https://github.com/logstash-plugins/logstash-patterns-core/blob/master/CONTRIBUTING.md#contribution-steps).
What the next step to merge that pull request?

**Comment by purbon** [comment]
@Hubbitus thanks for taking care of the CLA, will make sure to fix the broken link. Last step to merge this, is to add a few test. You can take inspiration on any of the test present at https://github.com/logstash-plugins/logstash-patterns-core/blob/master/spec/patterns/mongodb_spec.rb. This is very nice and let us be sure patterns behave as expected. Feel free to ask any question you've while adding test, more than open to help.

**Comment by purbon** [comment]
Hi, speaking about the link. Could you point me to the place where you
found it? I was investigating and the links I found send me to
https://github.com/elastic/logstash/blob/master/CONTRIBUTING.md#contribution-steps.
Looking forward to remove the 404.

/cheers

**Comment by Hubbitus** [comment]
I'm not familiar with ruby unfortunately. But it may be interesting. Do you have some guide or how to how setup test environment for writing that test? Detailed manuals?

**Comment by purbon** [comment]
Moin @Hubbitus , do you have a working ruby env setup? if not I recommend investigating what rvm [1] or rbenv [2] are. Then when you have a working ruby env, to start writing and running the test you can:

* Run ```bundle install``` to get all dependencies in.
* Add the test you aim to, you can probably copy how for example the syslog_specs [3] were added.
* Run the specs/test by ``` bundle exec rspec```

This is a super fast tutorial, feel free to ask more detailed questions.

Otherwise we can also work together, feel free to describe the test in plain english here and I can set you up the context necessary.

Looking forward to it,

[1] https://rvm.io/
[2] https://github.com/sstephenson/rbenv
[3] https://github.com/logstash-plugins/logstash-patterns-core/blob/master/spec/patterns/syslog_spec.rb

**Comment by Hubbitus** [comment]
What kind of ruby should I use? Jruby?

```
$ RUBYPICK=_mri_ bundle install
Your Gemfile lists the gem logstash-filter-grok (>= 0) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Fetching gem metadata from https://rubygems.org/..........
Fetching additional metadata from https://rubygems.org/..
Resolving dependencies...
Could not find gem 'logstash-devutils (>= 0) ruby' in the gems available on this machine.
```

```
$ RUBYPICK=_jruby_ bundle install
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true -Dsun.java2d.pmoffscreen=false -XX:+UseCompressedOops -XX:+DoEscapeAnalysis -XX:+AggressiveOpts -XX:+EliminateLocks -XX:+UseParallelGC -XX:+UseNUMA -XX:+TieredCompilation
Ignoring json-1.8.2 because its extensions are not built.  Try: gem pristine json --version 1.8.2
Your Gemfile lists the gem logstash-filter-grok (>= 0) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Ignoring json-1.8.2 because its extensions are not built.  Try: gem pristine json --version 1.8.2
LoadError: no such file to load -- jopenssl/load
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/jruby/lib/ruby/shared/openssl.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/security.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/security.rb:11
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/package.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/package.rb:43
          (root) at /usr/share/rubygems/rubygems/dependency_installer.rb:1
          (root) at /usr/share/rubygems/rubygems/dependency_installer.rb:3
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/installer.rb:1
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/installer.rb:2
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli/install.rb:1
             run at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli/install.rb:79
         install at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli.rb:145
             run at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor/command.rb:27
  invoke_command at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor/invocation.rb:126
        dispatch at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor.rb:359
            load at org/jruby/RubyKernel.java:1087
          (root) at /bin/bundle:23
```

**Comment by purbon** [comment]
@Hubbitus I have no experience with rubypick, but the only "requirement" to run this project and hole LS should be jruby should be enough. In my setup I use rbenv, but been using rvm since also a lot of time.

The error I see in your stacktrace is that jopenssl looks like to be missing in your fedora, does this comes out of another package?

**Comment by Hubbitus** [comment]
What is jopenssl? Should not it be pulled as dependency in bundle install run too?

**Comment by jordansissel** [comment]
> What is jopenssl? Should not it be pulled as dependency in bundle install run too?

The error you got seems to come from bundler itself. I don't know what RUBY_PICK is, but I don't have this problem under rvm :\ Sorry I can't be more helpful.

**Comment by Hubbitus** [comment]
https://github.com/fedora-ruby/rubypick
I think it just pick need ruby version.

**Comment by Hubbitus** [comment]
I do not see rvm in Fedora.

**Comment by Hubbitus** [comment]
What bundler version should be used?

**Comment by Hubbitus** [comment]
Ok.
I've  remove all ruby '*ruby*', then instal jruby and rubygem-bundler. So I assume have only jruby:
$ ruby --version
jruby 1.7.19 (1.9.3p551) 2015-04-16 fffffff on OpenJDK 64-Bit Server VM 1.8.0_60-b27 +jit [linux-amd64]

But error the same:
```
$ bundle install
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on -Dswing.aatext=true -Dsun.java2d.pmoffscreen=false -XX:+UseCompressedOops -XX:+DoEscapeAnalysis -XX:+AggressiveOpts -XX:+EliminateLocks -XX:+UseParallelGC -XX:+UseNUMA -XX:+TieredCompilation
Ignoring json-1.8.2 because its extensions are not built.  Try: gem pristine json --version 1.8.2
Your Gemfile lists the gem logstash-filter-grok (>= 0) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Ignoring json-1.8.2 because its extensions are not built.  Try: gem pristine json --version 1.8.2
LoadError: no such file to load -- jopenssl/load
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/jruby/lib/ruby/shared/openssl.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/security.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/security.rb:11
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/package.rb:1
         require at org/jruby/RubyKernel.java:1071
         require at /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54
          (root) at /usr/share/rubygems/rubygems/package.rb:43
          (root) at /usr/share/rubygems/rubygems/dependency_installer.rb:1
          (root) at /usr/share/rubygems/rubygems/dependency_installer.rb:3
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/installer.rb:1
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/installer.rb:2
          (root) at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli/install.rb:1
             run at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli/install.rb:79
         install at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/cli.rb:145
             run at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor/command.rb:27
  invoke_command at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor/invocation.rb:126
        dispatch at /usr/share/gems/gems/bundler-1.7.8/lib/bundler/vendor/thor.rb:359
            load at org/jruby/RubyKernel.java:1087
          (root) at /bin/bundle:23
```
